### ISSUE TITLE
bamsurgeon mutation rate does not match set VAF

### ISSUE THREAD

**Issue by dmarron** [adamewing/bamsurgeon#136]
I ran bamsurgeon v1.2 with a VAF set to 0.35 for all my snvs, but when I look at my mutated bam in IGV, I see calls with far fewer mutations than 0.35 of my reads.
For example, when I run bamsurgeon with a single snv from C to A at:
chr1    1374138        1374138        0.35    A
I see 7 A alleles and 63 C alleles in IGV (only about 10% mutated).
However, the truth vcf shows me:
chr1    1374138 .       C       A       100     PASS    SOMATIC;VAF=0.344262295082;DPR=22.0     GT      0/1
And claims that it mutated with a VAF of 0.344.  Also, the log files shows me:
INFO 2019-07-22 12:46:26,421 haplo_chr1_1374138_1374138 wrote: 61, mutated: 21
So both the log file and the truth vcf show that bamsurgeon thinks it mutated 21 A reads, but the final bam only appears to have 7 A reads.
![image](https://user-images.githubusercontent.com/394935/61650202-2b604980-ac81-11e9-8b24-e61b8b24ef8a.png)
Also, when I look at the read sequences in the bam files before and after bamsurgeon, only seven of the reads have different (mutated) sequences after running bamsurgeon

**Comment by adamewing** [comment]
Do the majority of your SNVs have VAFs << 0.35? Here are a few ideas to help troubleshoot:

0. Make a small .bam file (e.g. this variant +/- 5kbp) and make a variant file containing only this variant and re-run. Verify that the output VAF is still not ~0.35.

1. Are you using the same aligner for realignment as was used in generating the original bam file? (see `--aligner` options)

2. You can use `--tagreads` so mark reads that addsnv has used (adds a 'BS' tag) which will help you track them down in the output bam. Sometimes in cases like this the addition of the variant changes the mapping location of the read - you can determine whether this is the case via the tags. Also the reads will be in the .log files in the log directory output from the run.

**Comment by dmarron** [comment]
I originally ran bamsurgeon with a list of 1,000 SNVs, each with VAF set to 0.35.  I saw several snvs that had many fewer mutations than expected.  This snv at chr1:1374138 is highlighted because it is one of the most clear cases.
0.  I made a small bam file and made a variant file with only this variant, but I got the same result (only 7 actually mutated reads or 10% VAF).
1. Yes, I am using bwa mem to align and used --aligner mem as a parameter
2. I haven't run it with --tagreads, but I did look at every read covering the snv position in the bam file before and after bamsurgeon was run and confirmed that the mapping location did not change for any of the reads even after mutation and that only 7 reads had their sequence changed.
If you would like I would be happy to share the small bam file, log files, etc.

**Comment by adamewing** [comment]
Sure, send me the bam surrounding the site and the input variant file, adam.ewing@gmail.com, thanks!

**Comment by dmarron** [comment]
I just wanted to confirm that you got the email I sent you with the bam surrounding the site and the input variant file.  Thanks!

**Comment by adamewing** [comment]
I did - sorry I haven't bug into it yet. I did just have a look and I think I'll need the bam segment from chr1:1369138-1379138 (i.e. -/+ 5kbp of the snv), otherwise I lose too many reads because the read pair isn't intact and can't exactly replicate the problem. Would you be able to send that? Thanks!

**Comment by adamewing** [comment]
Closing, may have been solved via email - this turned out to be an issue with unpaired reads in a paired-end .bam file.

**Comment by robinycfang** [comment]
Hi @adamewing sorry to dig out this ancient issue, but I encountered the same issue when using BAMsurgeon to create SNVs with paired ended BAM. I used bwa mem for both the original BAM file and BAMsurgeon, but I got inconsistent VAF between the output VCF from BAMsurgeon and IGV. It looks like all the reads in the mutated sites became single ended reads somehow after using BAMsurgeon. Wondering how the issue was resolved?
